controllers/Weathers.js

```diff
@@ -0,0 +1,44 @@
+const BaseController = require('./Base');
+const View = require('../views/Base');
+const weatherUtils = require('../models/weather-utils.js');
+
+const ZONE_CODE_LENGTH = 10;
+const REQ_DATE_LENGTH = 8;
+
+module.exports = BaseController.extend({
+	name: "Weathers",
+	run: function(req, res, next) {
+		const zoneCode = req.params.zoneCode;
+		const reqDate = req.params.date
+		if (!isValidZoneCode(zoneCode) || !isValidReqDate(reqDate)) {
+			res.send("Invalid zone code or date");
+			return;
+		}
+		const date = convertDateToDbForm(reqDate);
+		weatherUtils.loadWeather(date, zoneCode).then(function(err, weather) {
```

Comment:
**weatherUtils.loadWeather(date, zoneCode)** function will return weather only(err will be thrown to **catch** callback.

You should remove err from **function(err, weather)**